Python: feat(devui): enable manual workflow triggers with comprehensive type support

ChrisMcKee1 opened the pull request:
Enable manual triggering of workflows in DevUI by adding full JSON Schema type support, including Union types and None/null handling for workflows with optional or no inputs.

## Key Feature - Manual Workflow Triggers
- Workflows with no inputs (type: null) can now be triggered directly
- Union types (X | None) properly detected via anyOf schema field
- Optional inputs with defaults enable single-click execution
- Smart run button adapts to input requirements automatically

## Type System Enhancements
- Support all primitive types: string, integer, number, boolean, array
- Handle null type for input-free workflows
- Detect optional types via anyOf and default: null patterns
- Type-aware array inputs with item type labels (strings, integers, numbers)
- Enhanced boolean fields with visual state indicators (true/false)
- Dedicated integer/number handlers with proper step values and validation

## UX Improvements
- Smart run button logic auto-detects input requirements
- Button text adapts: "Run Workflow" (no input) vs "Configure & Run" (has input)
- Play icon for direct execution, Settings icon for configuration needed
- Type-specific placeholders and validation feedback
- Proper form initialization with type-appropriate defaults

## Technical Implementation
- Extended JSONSchema interfaces with null type and anyOf for Union support
- Type guards distinguish primitive, object, and null inputs
- Enhanced canSubmit validation handles all type combinations and optionality
- Form validation properly treats None/null as valid for optional fields

ChrisMcKee1 committed:
feat(devui): enable manual workflow triggers with comprehensive type support

Enable manual triggering of workflows in DevUI by adding full JSON Schema type support,
including Union types and None/null handling for workflows with optional or no inputs.

Key Feature - Manual Workflow Triggers:
- Workflows with no inputs (type: null) can now be triggered directly
- Union types (X | None) properly detected via anyOf schema field
- Optional inputs with defaults enable single-click execution
- Smart run button adapts to input requirements automatically

Type System Enhancements:
- Support all primitive types: string, integer, number, boolean, array
- Handle null type for input-free workflows
- Detect optional types via anyOf and default: null patterns
- Type-aware array inputs with item type labels (strings, integers, numbers)
- Enhanced boolean fields with visual state indicators (true/false)
- Dedicated integer/number handlers with proper step values and validation

UX Improvements:
- Smart run button logic auto-detects input requirements
- Button text adapts: "Run Workflow" (no input) vs "Configure & Run" (has input)
- Play icon for direct execution, Settings icon for configuration needed
- Type-specific placeholders and validation feedback
- Proper form initialization with type-appropriate defaults

Technical Implementation:
- Extended JSONSchema interfaces with null type and anyOf for Union support
- Type guards distinguish primitive, object, and null inputs
- Enhanced canSubmit validation handles all type combinations and optionality
- Form validation properly treats None/null as valid for optional fields

markwallace-microsoft commented:
<!-- Pytest Coverage Comment: python-test-coverage-report -->
<a href="https://github.com/microsoft/agent-framework/blob/60697cf8a74ba9fd42d37058c2cd8d740dfa2c6b/README.md"><img alt="Python Test Coverage" src="https://img.shields.io/badge/Python Test Coverage-83%25-green.svg" /></a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>12293</b></td><td><b>2009</b></td><td><b>83%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1401 | 98 :zzz: | 0 :x: | 0 :fire: | 26.958s :stopwatch: |

ChrisMcKee1 committed:
chore: update Python dependencies to match main branch

- tomli: 2.2.1 -> 2.3.0
- uv: 0.8.23 -> 0.8.24
- virtualenv: 20.34.0 -> 20.35.3
- yarl: 1.21.0 -> 1.22.0

ChrisMcKee1 committed:
style(devui): apply ruff auto-fixes for code formatting

- Remove trailing whitespace
- Normalize quote style to double quotes
- Replace elif with if for clearer control flow
- Use direct attribute access for types.UnionType

victordibia commented on python/packages/devui/.gitignore:
```diff
@@ -17,3 +17,6 @@ __pycache__/
 .DS_Store
 Thumbs.db
+
+# Frontend build artifacts (generated by npm run build)
```
```suggestion
# Frontend build artifacts (generated by npm run build)
```
```suggestion

```

ChrisMcKee1 committed:
refactor(devui): address code review feedback

- Fix variable shadowing: rename 'origin' to 'list_origin' for list type checks
- Revert .gitignore changes: UI builds are intentionally bundled with source
- Build and commit frontend bundle (726.91 kB, within <800kb requirement)
- Bundle includes updated workflow input form and view components

ChrisMcKee1 committed:
test(devui): add unit tests for None and Optional type schema generation

ChrisMcKee1 committed:
fix: update microsoft-agents dependencies to v0.5.0 to avoid packaging bug in v0.5.1

eavanvalkenburg commented:
@ChrisMcKee1 @victordibia do we still need this work? and if so can we get it back in shape?
